### PR TITLE
feat: Add get for custom org repo role

### LINKED ISSUE #3370: Add missing function to get a custom repository role for an org
The function to [Get a custom repository role](https://docs.github.com/en/enterprise-cloud@latest/rest/orgs/custom-roles?apiVersion=2022-11-28#get-a-custom-repository-role) for an organization is currently missing. I'd expect to see a `GetCustomRepoRole` function.

### PR DESCRIPTION
**Pull request by stevehipwell**
Resolves #3370 

This PR adds the missing `GetCustomRepoRole` function.

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @stevehipwell !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

(Please use regular push (not force-push) in this repo because we always use "squash & merge" anyway and it is much easier for reviewers to see what has changed since the last look... thank you!)

**Comment by stevehipwell** [comment]
RE force push, I'll try and do that next time but it's muscle memory at this point so apologies in advance. I think the desktop GH interface now shows diffs for force push events, but I'm currently on mobile so can't check.

**Comment by stevehipwell** [comment]
@gmlewis I've rebased this PR, is there anything outstanding on it?

**Comment by gmlewis** [comment]
> @gmlewis I've rebased this PR, is there anything outstanding on it?

Only a second LGTM+Approval from any other contributor to this repo. 
All contributors are volunteers so now we patiently wait for a second review. If you have coworkers who can review, please welcome their feedback. Thank you!

**Comment by stevehipwell** [comment]
@gmlewis what is the requirement for the second approval? Someone who's already contributed or one of the maintainers or something else?

**Comment by gmlewis** [comment]
> @gmlewis what is the requirement for the second approval? Someone who's already contributed or one of the maintainers or something else?

We have never put any restrictions on this. Our official thoughts are detailed in CONTRIBUTING.md. In my personal opinion, we welcome contributions from anyone who is interested in the Go programming language.

**Comment by stevehipwell** [comment]
@gmlewis my question above was about where the second set of eyes to approve a PR needs to come from? Do you need another maintainer or just another set of eyes from somewhere?

**Comment by gmlewis** [comment]
> @gmlewis my question above was about where the second set of eyes to approve a PR needs to come from? Do you need another maintainer or just another set of eyes from somewhere?

I apologize that my answer was not clear, as I was attempting to answer this question.
The answer is the latter, preferably someone who is interested in the Go programming language.

**Comment by gmlewis** [comment]
@stevehipwell - can you please make a concerted effort to stop force-pushing in this repo, as now I must re-read the whole PR because I cannot see what changed since the last review.

**Review by gmlewis — APPROVED** [review]

**Comment by stevehipwell** [comment]
@gmlewis I'm just rebasing onto the min branch, my code isn't changing and GitHub shouldn't be clearing your approval. I can stop doing that, but you'll need to do it before the automation runs and the PR is merged to get a linear commit history (and in some cases GitHub can't do it in the UI).

**Review by tomfeigin — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @tomfeigin !
Merging.

**Pull request merged**
